js/piwik.js

```diff
@@ -184,7 +184,9 @@ if (typeof window.Matomo !== 'object') {
 
             trackerIdCounter = 0,
 
-            isPageUnloading = false;
+            isPageUnloading = false,
+
+            javaScriptErrors = [];
```

Comment:
Hey @diosmosis, I added a test. 

But now I have a question about this line. First of all, to make the test work, I needed to add a new method, to get this `javaSriptErrors`. That's fine I think.

I'm not sure this `javaSrciptErrors` is actually in a good place. Shouldn't it be around line 2388, inside the Matomo Tracker Class?

What's the difference and which one is better?
Thanks, Zoltan

Reply:
@flamisz if you add it to 2388, then there will be one array per tracker. It's possible for users to create multiple trackers on the same website/webpage that track to different matomo's or different idsites (for example). Right now the array is "global". I'm not sure there's a benefit to either way, since in either case we want to skip the same javascript errors. I guess, maybe if a new tracker is created later on in the user's visit (say for a single page app where the visit time can be long), and a similar javascript error comes up, maybe we still want to track that one? I'm not sure about the use case, maybe @tsteur has an opinion.

Reply:
Actually I guess it makes sense that we'd want to send the javascript error if the tracker hasn't seen it before, not if piwik.js hasn't seen it before. I think it makes sense to move the array if you can. Good call.

Reply:
great, that's what I thought. I move it and check the tests again if need to be updated or not. Thanks